interfaces/builtin/kubernetes_support.go

```diff
@@ -133,26 +133,26 @@ deny ptrace (trace) peer=unconfined,
 # kubelet calls out to systemd-run for some mounts, but not all of them and not
 # unmounts...
 capability sys_admin,
-mount /var/snap/@{SNAP_INSTANCE_NAME}/common/{,**} -> /var/snap/@{SNAP_INSTANCE_NAME}/common/{,**},
-mount options=(rw, rshared) -> /var/snap/@{SNAP_INSTANCE_NAME}/common/{,**},
+mount /var/snap/@{SNAP_INSTANCE_NAME}/*/{,**} -> /var/snap/@{SNAP_INSTANCE_NAME}/*/{,**},
+mount options=(rw, rshared) -> /var/snap/@{SNAP_INSTANCE_NAME}/*/{,**},
 
 /{,usr/}bin/mount ixr,
 /{,usr/}bin/umount ixr,
 deny /run/mount/utab rw,
-umount /var/snap/@{SNAP_INSTANCE_NAME}/common/**,
+umount /var/snap/@{SNAP_INSTANCE_NAME}/*/**,
 `
 
 const kubernetesSupportConnectedPlugAppArmorKubeletSystemdRun = `
   # kubelet mount rules
   capability sys_admin,
   /{,usr/}bin/mount ixr,
-  mount fstype="tmpfs" tmpfs -> /var/snap/@{SNAP_INSTANCE_NAME}/common/**,
+  mount fstype="tmpfs" tmpfs -> /var/snap/@{SNAP_INSTANCE_NAME}/*/**,
   deny /run/mount/utab rw,
 
   # For mounting volume subPaths
-  mount /var/snap/@{SNAP_INSTANCE_NAME}/common/{,**} -> /var/snap/@{SNAP_INSTANCE_NAME}/common/{,**},
-  mount options=(rw, remount, bind) -> /var/snap/@{SNAP_INSTANCE_NAME}/common/{,**},
-  umount /var/snap/@{SNAP_INSTANCE_NAME}/common/**,
+  mount /var/snap/@{SNAP_INSTANCE_NAME}/*/{,**} -> /var/snap/@{SNAP_INSTANCE_NAME}/*/{,**},
+  mount options=(rw, remount, bind) -> /var/snap/@{SNAP_INSTANCE_NAME}/*/{,**},
+  umount /var/snap/@{SNAP_INSTANCE_NAME}/*/**,
```

Comment:
again this seems like it would allow an individual revision to modify or do things to a different revision of the snap

Reply:
Yes, see previous comments.